libs/agno/agno/agent/agent.py

```diff
@@ -336,6 +341,16 @@ def __init__(
         self.memory = memory
         self.add_history_to_messages = add_history_to_messages
         self.num_history_responses = num_history_responses
+        self.num_history_runs = num_history_runs
+
+        if num_history_responses != 3:
```

Comment:
I'd rather make `num_history_responses` None by default at the top, then check here if it is set at all. If it is set, then set num_history_runs from it. So then you only do something if it is set at all.